test/extended/specs/gatewayapi.go

```diff
@@ -0,0 +1,157 @@
+package specs
+
+import (
+	"context"
+	"strings"
+
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+	"github.com/openshift/cluster-ingress-operator/test/extended/framework"
+
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	admissionapi "k8s.io/pod-security-admission/api"
+)
+
+var _ = g.Describe("[sig-network][OCPFeatureGate:GatewayAPI][Feature:Router][apigroup:gateway.networking.k8s.io]", func() {
+	defer g.GinkgoRecover()
+
+	var (
+		oc  *framework.CLI
+		err error
+	)
+
+	g.BeforeEach(func() {
+		oc, err = framework.NewCLI("gateway-api", admissionapi.LevelBaseline, framework.Get().K8sClient, framework.Get().RestCfg, framework.Get().IsOpenShift)
```

Comment:
while it may be tempting to abstract this and add this whole "framework.XXX" inside the client creation, I would prefer to receive these as parameters of NewCLI so this can be eventually reused